engine/src/damage/damageable_layer.h

```diff
@@ -0,0 +1,66 @@
+#ifndef DAMAGEABLELAYER_H
+#define DAMAGEABLELAYER_H
+
+#include <vector>
+
+#include "facet_configuration.h"
+#include "core_vector.h"
+#include "health.h"
+
+/**
+ * @brief The DamageableLayer class represents an object to be damaged.
+ * This can be shields, armor, hull or subsystem.
+ * A layer can be regenerative (e.g. shields) or not (pretty much everything else).
+ */
+struct DamageableLayer
+{
+    int layer_index;
```

Comment:
it would be better if this was an enum

Reply:
Not in this case. Remember that lib_damage doesn't understand hull, armor and shield. In theory you could use it in some other way - say with multiple layers of armor.